src/main/java/net/atos/zac/zoeken/IndexeerService.java

```diff
@@ -82,89 +86,95 @@ public IndexeerService() {
-        return new HerindexerenInfo(add, update, delete);
+        final int removeCount = countZoekIndexEntities(type, IndexStatus.REMOVE);
+        final int addCount = countZoekIndexEntities(type, IndexStatus.ADD);
+        final int updateCount = countZoekIndexEntities(type, IndexStatus.UPDATE);
+        LOG.info("[%s] Herindexeren gereed".formatted(type.toString()));
+        return new HerindexerenInfo(addCount, updateCount, removeCount);
     }
 
-    public int indexeer(int batchGrootte, ZoekObjectType type) {
+    public int indexeer(final int batchGrootte, final ZoekObjectType type) {
         if (isHerindexeren(type)) {
             LOG.info("[%s] Wachten met indexeren, herindexeren is nog bezig".formatted(type.toString()));
             return batchGrootte;
         }
-        final int count = countEntities(type);
+        final int count = countZoekIndexEntities(type);
         if (count == 0) {
             return 0;
         }
         LOG.info("[%s] aantal te indexeren: %d".formatted(type.toString(), count));
         final List<ZoekIndexEntity> entities = listEntities(type, batchGrootte);
-        final List<ZoekObject> addList = new ArrayList<>();
-        final List<String> deleteList = new ArrayList<>();
+        final List<ZoekObject> addOrUpdateList = new ArrayList<>();
+        final List<String> removeList = new ArrayList<>();
         final AbstractZoekObjectConverter<? extends ZoekObject> converter = getConverter(type);
         entities.forEach(zoekIndexEntity -> {
             try {
                 switch (IndexStatus.valueOf(zoekIndexEntity.getStatus())) {
-                    case ADD, UPDATE -> addList.add(converter.convert(zoekIndexEntity.getObjectId()));
-                    case REMOVE -> deleteList.add(zoekIndexEntity.getObjectId());
+                    case ADD, UPDATE -> addOrUpdateList.add(converter.convert(zoekIndexEntity.getObjectId()));
+                    case REMOVE -> removeList.add(zoekIndexEntity.getObjectId());
                     case INDEXED -> {
                         // skip..
                     }
                 }
             } catch (RuntimeException e) {
-                LOG.warning("[%s] Skipped %s: %s".formatted(type.toString(), zoekIndexEntity.getObjectId(), e.getMessage()));
+                LOG.warning("[%s] Skipped %s: %s".formatted(type.toString(), zoekIndexEntity.getObjectId(),
+                                                            e.getMessage()));
                 entityManager.remove(zoekIndexEntity);
             }
         });
-        addToSolr(addList);
-        deleteFromSolr(deleteList);
+        addToOrUpdateInSolr(addOrUpdateList);
+        removeFromSolr(removeList);
         entities.forEach(entity -> entityManager.remove(entity));
-        return countEntities(type);
+        return countZoekIndexEntities(type);
     }
 
-    private void processZaken() {
+    private void addZakenAsZaakZoekIndexEntitiesToBeAddedOrUpdated() {
```

Comment:
processZaken was misschien niet een sterke naam, maar als ik deze naam lees, dan snap in nog steeds niet wat het doet. readAllZakenAndMarkToAddOrUpdateInSolrIndex